module/settings.js

```diff
@@ -1,4 +1,153 @@
+import DG from "./config.js";
+
+const { ApplicationV2, HandlebarsApplicationMixin } = foundry.applications.api;
+
+const SettingForm = class extends HandlebarsApplicationMixin(ApplicationV2) {
+  constructor() {
+    super();
+  }
+
+  static _namespace;
+
+  static get namespace() {
+    return this.constructor._namespace
+  };
+
+  static get settings() {
+    return {}
+  }
+
+  static init() {
+    const settings = this.settings;
+    for (const setting of Object.keys(settings)) {
+      game.settings.register(DG.ID, setting, {
+        scope: "world",
+        config: false,
+        name: game.i18n.localize(`DG.Settings.${setting}.name`),
+        hint: game.i18n.localize(`DG.Settings.${setting}.hint`),
+        ...settings[setting],
+      });
+    }
+  }
+
+  static DEFAULT_OPTIONS = {
+    tag: "form",
+    id: `${DG.ID}-setting-form`,
+    classes: [DG.ID, "settings-menu"],
+    window: {title:  ``, resizable: true},
+    position: {width: 500, height: 'auto'},
+    actions: {},
+    form: {
+      handler: this.onSubmit,
+      submitOnChange: false,
+      closeOnSubmit: false
+    }
+  };
+
+  static PARTS = {
+    form: { template: `systems/${DG.ID}/templates/settings.hbs` },
+  }
+
+  static async onSubmit(event, form, formData) {
+    event.preventDefault();
+    const data = foundry.utils.expandObject(formData.object);
+
+    for (const key in data) {
+      game.settings.set(DG.ID, key, data[key]);
+    }
+  }
+
+  async _prepareContext(_options) {
+    let context = await super._prepareContext(_options);
+    context.settings = Object.entries(this.constructor.settings).reduce(function (obj, [key, setting]) {
+      obj[key] = {
+        ...setting,
+        key,
+        name: `DG.Settings.${key}.name`,
+        hint: `DG.Settings.${key}.hint`,
+        value: game.settings.get(DG.ID, key),
+        isSelect: !!setting.choices,
+        isNumber: setting.type === Number,
+        isString: setting.type === String,
+        isCheckbox: setting.type === Boolean,
+      };
+      return obj;
+    }, {});
+    return context;
+  }
+}
+
+class AutomationSettings extends SettingForm {
+
+  static _namespace = "automation";
+
```

Comment:
```suggestion
  static _namespace = "automation";

  static DEFAULT_OPTIONS = {
    window: { title: "Automation Settings" }
  };
```

We should change the settings title per subclass